From 10f94f74af8aab34527fae645bbf25b962e10f7f Mon Sep 17 00:00:00 2001
From: Dave Eddy <dave@daveeddy.com>
Date: Thu, 14 Feb 2019 13:03:19 -0500
Subject: [PATCH] OS-7581 metadata daemon not detecting VM status changes
 Reviewed by: Todd Whiteman <todd.whiteman@joyent.com> Approved by: Todd
 Whiteman <todd.whiteman@joyent.com>

---
 src/vm/lib/metadata/agent.js | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/vm/lib/metadata/agent.js b/src/vm/lib/metadata/agent.js
index 53ef0bee..ea5a1a5c 100644
--- a/src/vm/lib/metadata/agent.js
+++ b/src/vm/lib/metadata/agent.js
@@ -20,7 +20,7 @@
  *
  * CDDL HEADER END
  *
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  *
  *
  * # OVERVIEW
@@ -496,7 +496,9 @@ MetadataAgent.prototype.start = function start() {
         }
         // Only 1 state change event should be seen per vminfod event
         assert.equal(state.length, 1, 'multiple "state" changes seen');
-        state = state[0];
+        assert.object(state[0], 'state[0]');
+        assert.string(state[0].newValue, 'state[0].newValue');
+        state = state[0].newValue;
 
         // If a KVM zone stops while we're trying to reconnect to its metadata
         // socket, stop trying to reconnect.
-- 
2.21.0

